### PR TITLE
chore(rust): remove unused dependencies

### LINKED ISSUE #4403: Clean up dependency tree
- [ ] Why does `cargo build -p connlib-client-apple` build `firezone-relay`?
- [ ] Unused crate dependencies (see #4402)
- [ ] Unused features from dependencies

It might be more tedious but maybe it's a good idea to have more `target` sections for each platform in our `Cargo.toml`s so that we aren't inadvertently linking in libraries that aren't used. It can be tough to remove them later, even relying on the linker to do so.

During our hunting session for #4377, @ReactorScram and I found lots of cross-pollination of dependencies across platforms.

### PR DESCRIPTION
**Pull request by thomaseizinger**
These were all found by `cargo-udeps`.

Resolves: #4403.

### CODE REVIEW

File: rust/connlib/snownet/Cargo.toml [hunk 1]
```diff
@@ -5,9 +5,7 @@ version = "1.0.0"
 edition = "2021"
 
 [dependencies]
-anyhow = "1"
 boringtun = { workspace = true }
-firezone-relay = { workspace = true }
```
Comment (thomaseizinger) [inline rust/connlib/snownet/Cargo.toml:10]: Not sure what that was doing in here :sweat_smile:

**Comment by thomaseizinger** [comment]
I had to fix some warnings in our code ([1b3947f](https://github.com/firezone/firezone/pull/4475/commits/1b3947f65ac6121790b35752a8d744c2950bed6d)). Those showed up (and failed the build) because `cargo udeps` compiles the code using a nightly version and the nightly apparently has a better detection of unused imports that already come in via `*` imports.

I could move those to a separate PR but I am not sure it is worth it.

File: rust/gui-client/src-tauri/Cargo.toml [hunk 2]
```diff
@@ -46,7 +42,7 @@ tauri = { version = "1.6", features = [ "dialog", "shell-open-api", "system-tray
 tauri-runtime = "0.14.2"
 tauri-utils = "1.5.3"
 thiserror = { version = "1.0", default-features = false }
-tokio = { version = "1.36.0", features = ["signal", "time"] }
+tokio = { version = "1.36.0", features = ["signal", "time", "macros", "rt", "rt-multi-thread"] }
```
Comment (ReactorScram) [inline rust/gui-client/src-tauri/Cargo.toml:45]: Neat, it added those features that were accidentally pulled in from other packages in the workspace. That's been bugging me.

File: rust/gui-client/src-tauri/src/client/network_changes/linux.rs [hunk 3]
```diff
@@ -2,9 +2,6 @@
 
 use anyhow::Result;
 
-#[derive(thiserror::Error, Debug)]
-pub(crate) enum Error {}
```
Comment (ReactorScram) [inline rust/gui-client/src-tauri/src/client/network_changes/linux.rs:6]: I'm surprised this one didn't show up in Clippy.

**Review by ReactorScram — APPROVED** [review]
LGTM, I figure if you remove all that and it still compiles it's fine.

**Comment by jamilbk** [comment]
Thanks for taking over this one!

File: rust/gui-client/src-tauri/Cargo.toml (hunk 2 above)
Comment (thomaseizinger) [reply rust/gui-client/src-tauri/Cargo.toml:45]: I had to add those manually :)

**Pull request merged**